modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -3069,13 +3068,15 @@ public void setStreamerReceiverRunner(StreamerReceiverRunner runner) {
     }
 
     public Set<TableImpl> zoneTables(int zoneId) {
-        return tablesPerZone.computeIfAbsent(zoneId, id -> new HashSet<>());
+        // Using a concurrent set as a value as it can be read (and iterated over) without any synchronization by external callers.
+        return tablesPerZone.computeIfAbsent(zoneId, id -> ConcurrentHashMap.newKeySet());
```

Comment:
Shouldn't it be `getOrDefault` instead?

Reply:
If the zone has no tables, 10 calls with `getOrDefault()` will produce 10 instances of an empty set, while for `computeIfAbsent()` it will be just 1 instance. On the other hand, if we have a million empty zones, `getOrAbsent()` will hog less memory.

Both cases seem to be almost identical in real life. Does `getOrDefault()` have any significant advantages?

Reply:
> 10 calls with getOrDefault() will produce 10 instances of an empty set

Well, I expected `getOrDefault` to return `Set.of()` as default, so there will be only a single instance. It's strange that we currently populate the map during a `get` operation. Or do we somewhere rely on the fact that this is a shared set and some data will appear in it?

Reply:
No, we don't rely on set 'sharedness'. I agree, `getOrDefault()` is better, I changed the code.

I also made another change: public `zoneTables()` now makes a defensive copy.